A start on documenting the Facility Choice model (#1616)
ihmeuw/vivarium_research: docs/source/models/concept_models/vivarium_mncnh_portfolio/facility_model.rst

:orphan:

.. _2024_facility_model_vivarium_mncnh_portfolio:

Delivery Facility Choice Model
==============================

.. contents::
   :local:
   :depth: 2

Background
----------

> To capture the complex relationship between choice of delivery facility (home birth vs a facility with basic emergency obstetric and neonatal care [BEmONC] vs a facility with comprehensive care [CEmONC]), the belief about gestational age (believed pre-term vs believed full term), and the related factors of antenatal care (ANC), and low birth weight and short gestation (LBWSG) risk exposure, we will include two novel affordances in our simulation: (1) correlated propensities for ANC, home delivery, and LBWSG; and (2) conditional probabilities for home delivery that differ based on the believe pre-term/full-term status when labor begins.

Reviewer: ```suggestion
To capture the complex relationship between choice of delivery facility (home birth vs a facility with basic emergency obstetric and neonatal care [BEmONC] vs a facility with comprehensive care [CEmONC]), the belief about gestational age (believed pre-term vs believed full term), and the related factors of antenatal care (ANC), and low birth weight and short gestation (LBWSG) risk exposure, we will include two novel affordances in our simulation: (1) correlated propensities for ANC, home delivery, and LBWSG; and (2) conditional probabilities for home delivery that differ based on the believed pre-term/full-term status when labor begins.
```

    copula.rvs(10)

.. list-table:: Propensity Correlation Stand-in Parameters
   :header-rows: 1
   :widths: 20 20 20

   * - Factor A
     - Factor B
     - Correlation
   * - ANC Propensity
     - In-facility Birth Propensity
     - 0.50
   * - ANC Propensity
     - LBWSG Propensity
     - 0.10
   * - In-facility Birth Propensity
     - LBWSG Propensity
     - 0.10

Eventually we must put draws of consistent values in the artifact.

Special ordering of the categories
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Our method of inducing correlations using a Gaussian copula is equivalent to specifying the `polychoric correlation <https://en.wikipedia.org/wiki/Polychoric_correlation>`_ between the variables, and relies on having a known ordering of variable values.

When we calibrate the model, we will use an ordering of the LBWSG categories that makes them most likely to correlate with the ANC and home delivery propensities.  To achieve this, we will order them from lowest average RR to highest average RR (averaged across all draws).  Note that this is sex-specific, and based on the RRs for the early neonatal age group!  This ordering must also be used in the Risk component.

> We will also order the ANC and home delivery propensities from lowest to highest risk: some ANC < no ANC; and CEmONC < BEmONC < home birth.

Reviewer: I used the opposite ordering in my code, i.e., highest risk to lowest risk. I guess it should work out the same, but maybe it would be better to keep it consistent with my implementation? Also, highest risk to lowest risk matches GBD's usual convention for risk factors (i.e., cat1 is generally the highest risk category, and the TMREL is the highest numbered category by default).

Also, it seems like maybe we should be a bit more explicit about what "ordering the categories" or "ordering the propensities" means in the code. Namely, if the categories are ordered from highest risk to lowest risk as $c_1,\dotsc, c_n$, divide the unit interval $[0,1]$ into $n$ subintervals $I_1,\dotsc, I_n$ ordered from left to right, such that the length of $I_j$ is $P(c_j)$. Then a uniform propensity $p\in [0,1]$ corresponds to category $c_j$ precisely when $p\in I_j$. A picture would probably help.

Reviewer: > When we calibrate the model, we will use an ordering of the LBWSG categories that makes them most likely to correlate with the ANC and home delivery propensities.

Ok, I think now I see what you're saying here. It might be more accurate to say something like, "we will order the categories of all three variables (LBWSG, ANC, and facility) in such a way that their corresponding propensities are likely to be linearly correlated." But I don't know if that's necessarily any clearer.

Reviewer 2: I am not sure if this is an issue or not, but it seems possible to me that ordering may be draw-specific. Would this cause any issues? Should we order based on a specific draw in that case?

Author: I've made some edits to try to clarify this

Reviewer 2: Gotcha, makes sense and sounds good.

In addition to correlation, we posit that a belief about preterm status is influential in the decision to have a home delivery.  We will model this as a conditional probability of home delivery given a belief about preterm status.  Although deriving consistent values for these conditional probabilities is complex, and described in the final section of this page, *using* the conditional probabilities is simple: simply select the facility type with :math:`\text{Pr}[\text{facility}\mid\text{believed preterm}]` and :math:`\text{Pr}[\text{facility}\mid\text{believed fullterm}]` for the corresponding cases.

.. list-table:: Conditional Probability Stand-in Parameters
   :header-rows: 1
   :widths: 20 20 20 20

   * - Belief
     - Home
     - BeMONC
     - CeMONC
   * - Term
     - 0.25
     - 0.10
     - 0.65
   * - Preterm
     - 0.05
     - 0.05
     - 0.90

Challenge of calibrating the model
----------------------------------

We have developed a nonlinear optimization model to find a consistent set of parameters for the Gaussian copula and the conditional probabilities.
It will be described in detail here.

Link to code implementing it, too.

> Range of propensity and probabilities (and probability gaps) that are consistent with existing data

Reviewer: What do you mean by "probability gaps"?

Author: I was thinking of the difference between the conditional probabilities conditioned on believed full term and believed preterm. I'll add some text to make it clearer.

:orphan:

.. _2024_facility_model_vivarium_mncnh_portfolio:

Delivery Facility Choice Model
==============================

.. contents::
   :local:
   :depth: 2

Background
----------

> To capture the complex relationship between choice of delivery facility (home birth vs a facility with basic emergency obstetric and neonatal care [BEmONC] vs a facility comprehensive care [CEmONC]), the belief about gestational age (believed pre-term vs believed full term), and the related factors of antenatal care (ANC), and low birth weight small for gestational age (LBWSG) risk exposure, we will include two novel affordances in our simulation: (1) correlated propensities for ANC, home delivery, and LBWSG; and (2) conditional probabilities for home delivery that differ based on the believe pre-term/full-term status when labor begins.

Reviewer: ```suggestion
To capture the complex relationship between choice of delivery facility (home birth vs a facility with basic emergency obstetric and neonatal care [BEmONC] vs a facility comprehensive care [CEmONC]), the belief about gestational age (believed pre-term vs believed full term), and the related factors of antenatal care (ANC), and low birth weight and short gestation (LBWSG) risk exposure, we will include two novel affordances in our simulation: (1) correlated propensities for ANC, home delivery, and LBWSG; and (2) conditional probabilities for home delivery that differ based on the believe pre-term/full-term status when labor begins.
```
Wrong name for the LBWSG risk

Coming up with values for these correlations and conditional probabilities that are consistent with GBD and external evidence is detailed at the end of this document.  But before we get to that complexity, let's start with how we will use these correlations and conditional probabilities in the simulation.

[[Causal model diagram goes here]]

> 

Reviewer: Somewhere, I think we should mention that the calibration procedure, and hence the values we're using here (i.e., the correlations and values of P(facility | believed preterm status)) depend critically on the implementations of other pieces of the model that are described elsewhere (most notably, choice of ultrasound type given ANC status, and deriving an estimated gestational age from the true gestational age).

Correlated propensities
-----------------------

This section describes how we will model an "instrisic correlation" of ANC, home delivery, and LBWSG.  In short, we will use a Gaussian copula to model this, which has three parameters capturing the correlation between each pair of the three propensities.

> The motivation for these correlations is as follows: we hypothesize that there are important "common causes" that are not shown explicitly in the diagram above.  For example, having a home delivery and having no ANC visits might both be influenced by rurality --- if all health services are offered far away, it is logical that people will be able to access them less.

Reviewer: ```suggestion
The motivation for these correlations is as follows: We hypothesize that there are important "common causes" that are not shown explicitly in the diagram above.  For example, having a home delivery and having no ANC visits might both be influenced by rurality --- if all health services are offered far away, it is logical that people will be able to access them less.
```

Author: Since it is after a colon, I perfer to keep it lowercase.

Reviewer: [when to capitalize after a colon](https://www.google.com/search?q=when+to+capitalize+after+a+colon&rlz=1C5GCEM_enUS1023US1023&oq=when+to+capitalize+after+a+colon&gs_lcrp=EgZjaHJvbWUqDggAEEUYJxg7GIAEGIoFMg4IABBFGCcYOxiABBiKBTIICAEQABgWGB4yCAgCEAAYFhgeMggIAxAAGBYYHjIKCAQQABgKGBYYHjIICAUQABgWGB4yCAgGEAAYFhgeMggIBxAAGBYYHjIICAgQABgWGB4yCAgJEAAYFhge0gEIMTQwMmowajeoAgCwAgA&sourceid=chrome&ie=UTF-8)

> Similarly, it is likely that there are social exclusion factors causeing both exposure to LBWSG risk and lack of access to ANC and in-facility birth.

Reviewer 3: ```suggestion
Similarly, it is likely that there are social exclusion factors causing both exposure to LBWSG risk and lack of access to ANC and in-facility birth.
```

In a simulation model where we have not included scenarios that change these common-cause factors, we do not have to model their effects explicitly.
For our purposes, it is sufficient to capture the correlations between ANC, in-facility birth, and LBWSG risk exposure.

> In Vivarium, we use values selected uniformly at random from the interval [0,1], which we call propensities, to keep attributes like LBWSG and ANC calibrated at the population level.  This makes it straightforward to represent the correlation in our factors by generating correlated propensities. The `statsmodels.distributions.copula.api.GaussianCopula` implementation can make them::

Reviewer: ```suggestion
In Vivarium, we use values selected uniformly at random from the interval [0,1], which we call propensities, to keep attributes like LBWSG and ANC calibrated at the population level while reducing variance between scenarios at the simulant level.  This makes it straightforward to represent the correlation in our factors by generating correlated propensities. The `statsmodels.distributions.copula.api.GaussianCopula` implementation can make them::
```
Or something like that.

Reviewer: > This makes it straightforward to represent the correlation in our factors

Also, should this be "risk factors" instead of "factors"? Or maybe some other word?

Reviewer: We should say somewhere that we're using the fact that our three categorical variables are _ordered_ to define a sensible notion of correlation between them. I guess technically we could do the same thing with unordered categorical variables, but then the numerical values of the correlations would not be interpretable, and it would probably make the model harder to calibrate. In my nanosim, I'm ordering all categorical variables from "worst" to "best" (e.g., ANC0 < ANC1, and Home < BEmONC < CEmONC), so that larger propensities correspond to better outcomes . Keeping everything consistent makes it easier to reason about what the values of the correlations mean.

Actually, now that I'm saying it out loud (or in print, rather), I think we **need** to specify the ordering of the categories of _all_ the correlated variables, not just the LBWSG categories, so that the correlations in the sim will have the same meanings that they do in the calibration. If we just give the engineers the correlations without the category ordering, we're not guaranteed to produce the same results -- without knowing the ordering of the categories, the numerical values of the correlations are essentially meaningless.

I think we should also mention that our method of inducing correlations between ordinal variables using a Gaussian copula is equivalent to specifying the [polychoric correlation](https://library.virginia.edu/data/articles/understanding-polychoric-correlation#:~:text=Polychoric%20correlation%20is%20a%20measure,a%20bivariate%20standard%20normal%20distribution.) between the variables:

https://en.wikipedia.org/wiki/Polychoric_correlation

Instead of trying to estimate such a correlation like statisticians would usually do, we are building a generative model that explicitly induces some specified polychoric correlations.

    from statsmodels.distributions.copula.api import GaussianCopula
    copula = GaussianCopula([[1., .5, .1],
                             [.5, 1., .1],
                             [.1, .1, 1.]])
    copula.rvs(10)

.. list-table:: Propensity Correlation Stand-in Parameters
   :header-rows: 1
   :widths: 20 20 20

   * - Factor A
     - Factor B
     - Correlation
   * - ANC Propensity
     - In-facility Birth Propensity
     - 0.50
   * - ANC Propensity
     - LBWSG Propensity
     - 0.10
   * - In-facility Birth Propensity
     - LBWSG Propensity
     - 0.10

Eventually we must put draws of consistent values in the artifact.

Special ordering of LBWSG categories
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> When we calibrate the model, it will help to have an ordering of the LBWSG categories that makes them most likely to correlate with the ANC and home delivery propensities.  To achieve this, we will order them from lowest average RR to highest average RR (averaged across all draws).  [[Perhaps more details to come, such as code or an explicit table.]]

Reviewer: > it will help to have an ordering of the LBWSG categories that makes them most likely to correlate with the ANC and home delivery propensities

This phrase is confusing to me -- I'm not sure what you're trying to say.

Reviewer: > To achieve this, we will order them from lowest average RR to highest average RR (averaged across all draws). [[Perhaps more details to come, such as code or an explicit table.]]

We need to specify that this ordering is sex-specific (so there's one ordering for males and a different ordering for females), and uses the RRs from the **early neonatal** age group since we're interested in quantifying the risk _at birth_.

Author: See if my edits work better for you
